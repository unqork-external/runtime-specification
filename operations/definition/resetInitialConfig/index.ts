export * from './resetInitialConfig.model'
export * from './resetInitialConfig.options'
