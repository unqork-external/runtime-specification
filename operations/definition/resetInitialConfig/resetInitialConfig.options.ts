import { Description, Example, Required } from '@tsed/schema'

export class ResetInitialConfigOperationOptions {
  @Required()
  @Example('firstNameTextField', 'panel.*')
  @Description('targetKey refers to the intended target to reset to the initial config.')
  targetKey: string
}
