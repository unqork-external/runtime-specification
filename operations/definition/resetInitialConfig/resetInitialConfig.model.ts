import { DiscriminatorValue, Required } from '@tsed/schema'

import { ResetInitialConfigOperationOptions } from './resetInitialConfig.options'
import { Alpha } from '../../../../decorators'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.RESET_INITIAL_CONFIG)
@Alpha()
export class ResetInitialConfigOperation extends Operation<OperationTypes.RESET_INITIAL_CONFIG> {
  @Required()
  public type = OperationTypes.RESET_INITIAL_CONFIG as const

  @Required()
  public options: ResetInitialConfigOperationOptions = new ResetInitialConfigOperationOptions()
}
