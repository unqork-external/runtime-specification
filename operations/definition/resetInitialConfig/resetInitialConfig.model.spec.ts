import { ResetInitialConfigOperation } from './resetInitialConfig.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: ResetInitialConfig Tests', () => {
  it('ResetInitialConfig Model', () => {
    const { schema } = generateSchemaAndValidate(ResetInitialConfigOperation)

    expect(schema).toMatchSnapshot()
  })
})
