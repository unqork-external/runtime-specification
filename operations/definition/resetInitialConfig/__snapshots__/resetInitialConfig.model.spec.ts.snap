// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: ResetInitialConfig Tests ResetInitialConfig Model 1`] = `
{
  "definitions": {
    "ResetInitialConfigOperationOptions": {
      "properties": {
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "targetKey": {
          "description": "targetKey refers to the intended target to reset to the initial config.",
          "examples": [
            "firstNameTextField",
            "panel.*",
          ],
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/ResetInitialConfigOperationOptions",
    },
    "type": {
      "const": "RESET_INITIAL_CONFIG",
      "examples": [
        "RESET_INITIAL_CONFIG",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "type": "object",
}
`;
